### PR TITLE
chore: updates for 0.7.3

### PR DESCRIPTION
**Pull request by pameladelgado**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+-----
+
+This is a minor release that contains mainly improvements and fixes to the Renku CLI.
+Templates now use this new Renku CLI version ``0.12.2`` by default.
+
+New features
+~~~~~~~~~~~~
+
+- **Migrations** A ``migrationscheck`` command has been added.
+- **Git LFS** Large files are automatically added to LFS.
```
Comment (rokroskar) [inline CHANGELOG.rst:13]: This has actually been the case for some time - I would maybe rephrase this with "When a session is closed and work is automatically saved, large files are added to LFS according to the project's settings (see the [`lfs_threshold` configuration](https://renku-python.readthedocs.io/en/v0.12.0/commands.html#available-configuration-values))"

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+0.7.3
+-----
+
+This is a minor release that contains mainly improvements and fixes to the Renku CLI.
+Templates now use this new Renku CLI version ``0.12.2`` by default.
+
+New features
+~~~~~~~~~~~~
+
+- **Migrations** A ``migrationscheck`` command has been added.
```
Comment (rokroskar) [inline CHANGELOG.rst:12]: User's won't know what this means - we can say something like "Projects using older versions of renku can be automatically migrated."

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+- **Git LFS** Large files are automatically added to LFS.
+- **Core** Renku CLI commit messages are shorten to 100 characters for readability.
+
+
+Fixes
+~~~~~
+
+- **Auth credentials**: the way Renku environments handle oauth credentials has been improved.
```
Comment (rokroskar) [inline CHANGELOG.rst:20]:
```suggestion
- **Auth credentials**: the way Renku environments handle git credentials has been improved.
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -1,5 +1,27 @@
 .. _changelog:
 
+0.7.3
+-----
+
+This is a minor release that contains mainly improvements and fixes to the Renku CLI.
```
Comment (rokroskar) [inline CHANGELOG.rst:6]:
```suggestion
This is a minor release that contains mainly improvements and fixes to the Renku CLI and core service.
```

**Review by rokroskar — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst (hunk 2 above)
Comment (pameladelgado) [reply CHANGELOG.rst:12]: We had project migration before.
I think this is actually a new command that shows the state of a project, a user could run this inside an environment or locally to check the state. [ref issue](https://github.com/SwissDataScienceCenter/renku-python/pull/1761)

File: CHANGELOG.rst (hunk 1 above)
Comment (pameladelgado) [reply CHANGELOG.rst:13]: This makes more sense, I could not get that from the release notes. Thanks for the clarification!

Comment (rokroskar) [reply CHANGELOG.rst:13]: well technically we were asking for confirmation before and now it can be configured to just happen automatically. But I'm not sure a user will necessarily make that distinction

Comment (rokroskar) [reply CHANGELOG.rst:13]: argh I linked to the wrong docs - should be https://renku-python.readthedocs.io/en/v0.12.2/commands.html#available-configuration-values

File: CHANGELOG.rst [hunk 5]
```diff
@@ -1,5 +1,27 @@
...
+Templates now use this new Renku CLI version ``0.12.2`` by default.
+
+New features
+~~~~~~~~~~~~
+
+- **Migrations** A ``migrationscheck`` command has been added.
+- **Git LFS** Large files are automatically added to LFS.
+- **Core** Renku CLI commit messages are shorten to 100 characters for readability.
```
Comment (ableuler) [inline CHANGELOG.rst:14]:
```suggestion
- **Core** Renku CLI commit messages are shortened to 100 characters for readability.
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -1,5 +1,27 @@
...
+
+
+Fixes
+~~~~~
+
+- **Auth credentials**: the way Renku environments handle git credentials has been improved.
```
Comment (ableuler) [inline CHANGELOG.rst:20]:
```suggestion
- **Auth**: the way Renku environments handle the users GitLab credentials has been improved.
```

**Review by ableuler — CHANGES_REQUESTED** [review]
Some mini-suggestions regarding the wording.

File: CHANGELOG.rst (hunk 2 above)
Comment (pameladelgado) [reply CHANGELOG.rst:12]: I've changed this part in the new release notes

File: CHANGELOG.rst [hunk 7]
```diff
@@ -1,5 +1,73 @@
...
+- **Notifications** A notification system has been added so that users get notified for updates and events related to their projects.
+
+- **Project migration** An automatic migration to a new Renku version is possible. Information about the latest and current Renku CLI and template versions are displayed in the ``Status`` section of a project. Additionally, a ``migrationscheck`` command is available in Renku CLI.
+
+Improvements
+~~~~~~~~~~~~
+
+- **Dataset upload** When uploading files, a progress bar is displayed.
+
+- **Dataset visualization** When listing datasets, the description and author's list of a dataset are cropped. Full description and author list are still available on mouse hovering.
```
Comment (vfried) [inline CHANGELOG.rst:24]: For a better experience on dataset listing inside projects, description and authors have been cropped. Full description and authors are available when accessing the dataset.

File: CHANGELOG.rst [hunk 8]
```diff
@@ -1,5 +1,73 @@
...
+New features
+~~~~~~~~~~~~
+
+- **Dataset removal** A dataset can be now removed from a project from the UI. Note that if a removed dataset was part of a project with the Knowledge Graph activated, its metadata will not be removed.
+
+- **Pinned environments image** An image tag can be specified for launching environments. This can be done in the `renku configuration file <https://renku.readthedocs.io/en/latest/user/templates.html?highlight=.dockerignore#renku>`_.
+
+- **Notifications** A notification system has been added so that users get notified for updates and events related to their projects.
+
+- **Project migration** An automatic migration to a new Renku version is possible. Information about the latest and current Renku CLI and template versions are displayed in the ``Status`` section of a project. Additionally, a ``migrationscheck`` command is available in Renku CLI.
```
Comment (vfried) [inline CHANGELOG.rst:17]: - **Project migration** Project migration has been improved, allowing users to migrate the template, Dockerfile and renku version with just one click. Information about the latest and current Renku CLI and template versions are displayed in the ``Status`` section of a project. Additionally, a ``migrationscheck`` command is available in Renku CLI.

File: CHANGELOG.rst [hunk 9]
```diff
@@ -1,5 +1,73 @@
...
 
+0.7.3
+-----
+
+This release contains some very nice improvements to the file and datasets management and visualization as well as project migrations.
+
+New features
+~~~~~~~~~~~~
+
+- **Dataset removal** A dataset can be now removed from a project from the UI. Note that if a removed dataset was part of a project with the Knowledge Graph activated, its metadata will not be removed.
```
Comment (vfried) [inline CHANGELOG.rst:11]: I would remove this part...  Note that if a removed dataset was part of a project with the Knowledge Graph activated, its metadata will not be removed. 
 Since datasets can't be accessed anymore, from a user's perspective the datasets are gone.

Comment (jachro) [inline CHANGELOG.rst:11]: I'm just thinking if that could be a bit hard to understand for the users. I'd say that from a user perspective the data is gone as they cannot see it anymore anywhere. The project is still KG opted-in so no so sure if we need to say we soft-deleted it.

Comment (pameladelgado) [reply CHANGELOG.rst:11]: right, I thought the deleted dataset would still be displayed on data search. I'll rephrase this.

File: CHANGELOG.rst [hunk 10]
```diff
@@ -1,5 +1,73 @@
...
+
+- **Pinned environments image** An image tag can be specified for launching environments. This can be done in the `renku configuration file <https://renku.readthedocs.io/en/latest/user/templates.html?highlight=.dockerignore#renku>`_.
```
Comment (ciyer) [inline CHANGELOG.rst:13]: I suggest changing the wording:
```
A project can pin interactive environments to a specific image, independent of the content of the project. This can be useful for situations like courses where everyone should use the environment defined by the instructor.

To take advantage, see the `renku configuration file documentation <https://renku.readthedocs.io/en/latest/user/templates.html?highlight=.dockerignore#renku>`_.
```

File: CHANGELOG.rst [hunk 11]
```diff
@@ -1,5 +1,73 @@
...
+
+- **Notifications** A notification system has been added so that users get notified for updates and events related to their projects.
```
Comment (ciyer) [inline CHANGELOG.rst:15]: Let us leave this line out. We did introduce a client-side notification infrastructure, but it isn't very visible to users yet.

**Review by ciyer — CHANGES_REQUESTED** [review]
Thanks for compiling the Changelog from all the components! 

From the UI side, it looks very good. I made two suggestions -- one rewording, and one removal because the feature isn't yet very visible to users.

**Review by rokroskar — APPROVED** [review]

**Comment by rokroskar** [comment]
I just realized you downgraded the notebook version - was that on purpose?

**Comment by pameladelgado** [comment]
> I just realized you downgraded the notebook version - was that left on purpose?

Yes, because the tests are not passing with the last commit and it looks like it would take some time to fix that, so I've used a previous version of the notebooks that is compatible with the UI changes and for which the tests pass.

**Review by ciyer — APPROVED** [review]

**Comment by rokroskar** [comment]
@pameladelgado it's already been fixed by @ableuler 🤞

**Comment by rokroskar** [comment]
but it's ok for me to do this and roll out the new version of notebooks when it's fully ready

File: charts/renku/requirements.yaml [hunk 12]
```diff
@@ -2,27 +2,27 @@ dependencies:
...
 - name: renku-ui
   alias: ui
   repository: "https://swissdatasciencecenter.github.io/helm-charts/"
-  version: 0.11.2
+  version: 0.11.3
 - name: renku-notebooks
   alias: notebooks
   repository: "https://swissdatasciencecenter.github.io/helm-charts/"
-  version: 0.8.5
+  version: 0.8.5-c35d522
```
Comment (ableuler) [inline charts/renku/requirements.yaml:13]:
```suggestion
  version: 0.8.6
```
`0.8.6` has been fixed now.

**Review by ableuler — CHANGES_REQUESTED** [review]

**Review by ableuler — APPROVED** [review]
